src/main/java/com/shopify/minishopify/model/Product.java

```diff
@@ -23,31 +26,44 @@
 
     @ManyToOne(cascade = ALL)
     @JoinColumn(name = "shop_id", referencedColumnName = "id")
+    @JsonBackReference
     private Shop shop;
 
     @Column(name = "name")
+    @NotEmpty(message = "Product must have a name")
     private String name;
 
+    @Column(name = "description")
+    @NotEmpty(message =  "Product must have a description")
+    private String description;
+
     @Column(name = "price")
-    private float price;
+    @NotNull(message = "Product must have a price")
+    @Positive(message = "Price should be greater than 0")
+    private Float price;
 
     @Column(name = "quantity")
-    private int quantity;
+    @NotNull(message = "Product must have a quantity")
+    @Positive(message = "Quantity should be greater than 0")
+    private Integer quantity;
 
     @Column(name = "image")
-    private byte[] image;
+    @NotNull(message = "Product must have an image")
```

Comment:
I don't agree with this, unless we provide defaults.

Reply:
This is the point. Because we don't want to provide default therefore we require an image. This was a team decision.